src/Parser/IdentifierAndResource.php

```diff
@@ -168,35 +168,34 @@ public function getRelationships(): iterable
 
         $this->relationshipsCache = [];
 
-        $currentPath    = $this->getPosition()->getPath();
-        $nextLevel      = $this->getPosition()->getLevel() + 1;
+        $currentPath    = $this->position->getPath();
+        $nextLevel      = $this->position->getLevel() + 1;
         $nextPathPrefix = empty($currentPath) === true ? '' : $currentPath . PositionInterface::PATH_SEPARATOR;
-        foreach ($this->getSchema()->getRelationships($this->getData()) as $name => $description) {
+        foreach ($this->schema->getRelationships($this->data) as $name => $description) {
             assert($this->assertRelationshipNameAndDescription($name, $description) === true);
 
             [$hasData, $relationshipData, $nextPosition] = $this->parseRelationshipData(
-                $this->getFactory(),
-                $this->getSchemaContainer(),
-                $this->getType(),
+                $this->factory,
+                $this->schemaContainer,
+                $this->type,
                 $name,
                 $description,
                 $nextLevel,
                 $nextPathPrefix
             );
 
             [$hasLinks, $links] =
-                $this->parseRelationshipLinks($this->getSchema(), $this->getData(), $name, $description);
+                $this->parseRelationshipLinks($this->schema, $this->data, $name, $description);
 
-            $hasMeta = array_key_exists(SchemaInterface::RELATIONSHIP_META, $description);
```

Comment:
Meta can be any so `array_key_exists` should be used.

Reply:
Does `null` meta should use "hasMeta" flow?